### PR TITLE
CORE-6858 Uniqueness checker error handling refinement

### PR DESCRIPTION
**Pull request by relyafi**
Introduce new uniqueness check exception types to better support unexpected exceptions. See https://github.com/corda/corda-runtime-os/pull/2216 for full details.

### CODE REVIEW

File: data/avro-schema/src/main/resources/avro/net/corda/data/uniqueness/UniquenessCheckResultUnhandledExceptionAvro.avsc [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "type": "record",
+  "name": "UniquenessCheckResultUnhandledExceptionAvro",
+  "namespace": "net.corda.data.uniqueness",
+  "fields": [
```
Comment (r3hj) [inline data/avro-schema/src/main/resources/avro/net/corda/data/uniqueness/UniquenessCheckResultUnhandledExceptionAvro.avsc:5]: Are these fields defined somewhere or free to set any? Are `name` and `type` predefined keywords?

Comment (r3hj) [reply data/avro-schema/src/main/resources/avro/net/corda/data/uniqueness/UniquenessCheckResultUnhandledExceptionAvro.avsc:5]: After reading about Avro, the code/schema makes sense.

**Review by nkovacsx — DISMISSED** [review]
Looks good!

**Review by mnesbit — DISMISSED** [review]
OK

**Review by r3hj — DISMISSED** [review]
LGTM

**Review by mnesbit — APPROVED** [review]
OK

**Pull request merged**